### PR TITLE
Migrate `raw-dylib-inline-cross-dylib` and `raw-dylib-custom-dlltool` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: i686-mingw

### CODE REVIEW

File: tests/run-make/raw-dylib-inline-cross-dylib/Makefile [hunk 1]
```diff
@@ -1,30 +0,0 @@
-# Regression test for calling an inline function that uses a raw-dylib function.
-
-# only-windows
-
-include ../tools.mk
-
-# We'd be using the llvm-objdump instead of the system objdump to ensure compatibility
-# with the LLVM bitcode generated by rustc but on Windows  piping/IO redirection under MSYS2 is wonky with llvm-objdump.
-OBJDUMP = objdump
```
Comment (jieyouxu) [inline tests/run-make/raw-dylib-inline-cross-dylib/Makefile:9]: Remark: this is cursed

**Review by jieyouxu — APPROVED** [review]
Thanks, r=me if try jobs come back green.

**Comment by Oneirical** [comment]
```
2024-07-08T16:50:45.5638275Z test failed: `output.txt` is different from `actual_output`
2024-07-08T16:50:45.5638627Z 
2024-07-08T16:50:45.5638764Z --- output.txt
2024-07-08T16:50:45.5639026Z +++ actual_output
2024-07-08T16:50:45.5639313Z @@ -1,6 +1,6 @@
2024-07-08T16:50:45.5639594Z -extern_fn_1
2024-07-08T16:50:45.5639876Z -extern_fn_2 in extern_2
2024-07-08T16:50:45.5640214Z -extern_fn_2 in extern_1
2024-07-08T16:50:45.5640520Z -extern_fn_1
2024-07-08T16:50:45.5640804Z -extern_fn_2 in extern_2
2024-07-08T16:50:45.5641123Z -extern_fn_2 in extern_2
2024-07-08T16:50:45.5641395Z +extern_fn_1
2024-07-08T16:50:45.5641644Z +extern_fn_2 in extern_2
2024-07-08T16:50:45.5641923Z +extern_fn_2 in extern_1
2024-07-08T16:50:45.5642205Z +extern_fn_1
2024-07-08T16:50:45.5642446Z +extern_fn_2 in extern_2
2024-07-08T16:50:45.5642738Z +extern_fn_2 in extern_2
```

What sorcery is this?

**Comment by Oneirical** [comment]
TIL line endings are different on DOS and UNIX: https://www.cs.toronto.edu/~krueger/csc209h/tut/line-endings.html

@bors try

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Pull request merged**